src/graphics/rc_sprite.rs

```diff
@@ -0,0 +1,188 @@
+use crate::{
+    ffi::graphics as ffi,
+    graphics::{
+        Color, Drawable, FloatRect, IntRect, RcTexture, RenderStates, RenderTarget, Texture,
+        Transform, Transformable,
+    },
+    sf_box::SfBox,
+    system::Vector2f,
+};
+use std::{
+    cell::RefCell,
+    ptr::{self, NonNull},
+    rc::Weak,
+};
+
+#[derive(Debug)]
+pub struct RcSprite {
+    sprite: NonNull<ffi::sfSprite>,
+    texture: Weak<RefCell<SfBox<Texture>>>,
+}
+
+impl RcSprite {
+    #[must_use]
+    pub fn new() -> Self {
+        let sp = unsafe { ffi::sfSprite_create() };
+        Self {
+            sprite: NonNull::new(sp).expect("Failed to create Sprite"),
+            texture: Weak::new(),
+        }
+    }
+
+    fn texture_exists(&self) -> bool {
+        self.texture.strong_count() != 0
+    }
+
+    fn set_rc_texture(&mut self, texture: &RcTexture) {
+        self.texture = texture.downgrade()
+    }
+
+    #[must_use]
+    pub fn with_texture(texture: &RcTexture) -> RcSprite {
+        let mut sprite = Self::new();
+        sprite.set_texture(&texture, true);
+        sprite
+    }
+
+    #[must_use]
+    pub fn with_texture_and_rect(texture: &RcTexture, rect: IntRect) -> RcSprite {
+        let mut sprite = Self::with_texture(texture);
+        sprite.set_texture_rect(rect);
+        sprite
+    }
+
+    #[must_use]
```

Comment:
unwrap here should be safe as I had just set the texture in the `set_rc_texture` function